controllers/cloud.redhat.com/namespacereservation_controller.go

```diff
@@ -108,6 +108,16 @@ func (r *NamespaceReservationReconciler) Reconcile(ctx context.Context, req ctrl
 			return ctrl.Result{}, err
 		}
 		if r.Poller.namespaceIsExpired(expirationTS) {
+			err = DeleteSubscriptionPrometheusOperator(ctx, r.Client, res.Status.Namespace)
+			if k8serr.IsNotFound(err) {
```

Comment:
It looks like you're already doing this in the `DeleteSubscriptionPrometheusOperator` method. Do we want to do that again here?